src/commands/deploy/showCoreToolsWarning.ts

```diff
@@ -10,52 +10,52 @@ import { FuncVersion, funcVersionLink, tryParseFuncVersion } from "../../FuncVer
-                }
-                if (result === DialogResponses.dontWarnAgain) {
-                    await updateGlobalSetting(showCoreToolsWarningKey, false);
-                }
+                await showCoreToolsEOLWarning(context, localVersion);
             } else if (localVersion !== projectVersion) {
-                result = await showCoreToolsMismatchWarning(context, localVersion, projectVersion);
-                while (result === DialogResponses.learnMore) {
-                    await openUrl(funcVersionLink);
-                    result = await showCoreToolsMismatchWarning(context, localVersion, projectVersion);
-                }
-                if (result === DialogResponses.dontWarnAgain) {
-                    await updateGlobalSetting(showCoreToolsWarningKey, false);
-                }
+                await showCoreToolsMismatchWarning(context, localVersion, projectVersion, projectName);
             }
         }
     }
 }
 
+async function showCoreToolsWarningHelper(context: IActionContext, message: string): Promise<MessageItem> {
+    let result: MessageItem;
+    const showCoreToolsWarningKey: string = 'showCoreToolsWarning';
+    result = await context.ui.showWarningMessage(message, DialogResponses.learnMore, DialogResponses.dontWarnAgain);
+    while (result === DialogResponses.learnMore) {
+        await openUrl(funcVersionLink);
+        result = await context.ui.showWarningMessage(message, DialogResponses.learnMore, DialogResponses.dontWarnAgain);
+    }
+
+    if (result === DialogResponses.dontWarnAgain) {
+        await updateGlobalSetting(showCoreToolsWarningKey, false);
+    }
+    return result;
+}
 async function showCoreToolsEOLWarning(context: IActionContext, localVersion: FuncVersion): Promise<MessageItem> {
     const message = localize(
         'outdatedFunctionRuntime',
         'Your Azure Functions Core Tools Version ({0}) is past its end of life. Update to the latest version for the best experience.',
         localVersion
     );
-    return await context.ui.showWarningMessage(message, DialogResponses.learnMore, DialogResponses.dontWarnAgain)
+    return await showCoreToolsWarningHelper(context, message);
 }
 
-async function showCoreToolsMismatchWarning(context: IActionContext, localVersion: FuncVersion, projectVersion: FuncVersion): Promise<MessageItem> {
+async function showCoreToolsMismatchWarning(context: IActionContext, localVersion: FuncVersion, projectVersion: FuncVersion, projectName: string): Promise<MessageItem> {
     const message = localize(
         'mismatchedFunctionRuntime',
-        'Your Azure Functions Core Tools Version ({0}) does not match your Azure Functions runtime version ({1}). Update to the latest version for the best experience.',
+        'The local Azure Functions Core Tools Version ({0}) does not match the Azure Functions runtime version ({2}) of function app "{1}". Ensure that the versions match for the best experience.',
```

Comment:
nit: lol it's so confusing to have the numbers not be in order. That being said, you listed them correctly in the function call, so I won't bother you to change it 😋

Reply:
Lol I had it originally the way you had it in the comment, but decided to change it and forgot to change the numbers 🫠